test/verify/check-s4u-ssh

```diff
@@ -0,0 +1,95 @@
+@skipImage("freeipa not currently in testing", "debian-testing")
+class TestS4USsh(MachineCase):
+    provision = {
+        "0": {"address": "10.111.113.1/20", "dns": "10.111.112.100"},
+        "1": {"address": "10.111.113.2/20", "dns": "10.111.112.100"},
+        "services": {"image": "services", "memory_mb": 2048}
+    }
+
+    def setUp(self):
+        super().setUp()
+        self.machines['services'].execute("/run-freeipa")
+
+    def testBasic(self):
+        client_machine = self.machine
+        sshd_machine = self.machines["1"]
+        ipa_machine = self.machines["services"]
+
+        # set hostname for readable logs
+        client_machine.execute("hostnamectl set-hostname sshclient.cockpit.lan")
+        sshd_machine.execute("hostnamectl set-hostname sshserver.cockpit.lan")
+        self.login_and_go("/system")
+
+        # join both machines
+        wait(lambda: client_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        wait(lambda: sshd_machine.execute("nslookup -type=SRV _ldap._tcp.cockpit.lan"))
+        wait(lambda: client_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan"))
+        wait(lambda: sshd_machine.execute("echo foobarfoo | realm join -U admin cockpit.lan"))
+
+        # configure ipa
+        ipa_machine.execute("docker exec freeipa bash -c 'echo foobarfoo | kinit admin@COCKPIT.LAN'")
+        # add user to impersonate
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa user-add --first=user --last=user user'")
+        # ipa_machine.execute("echo foobarfoo | ipa user-mod user --password")
+        # add service which will be allowed to delegate creds
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-add cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN --ok-as-delegate=true --ok-to-auth-as-delegate=true'")
+        # Allow restrieval of service/host keytabs by admin
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-allow-retrieve-keytab --user=admin cockpitclient/sshclient.cockpit.lan'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa host-allow-retrieve-keytab --user=admin sshserver.cockpit.lan'")
+        # set up delegation rule
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationtarget-add cockpit-target'")
+
+        # replace with
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationtarget-add-member cockpit-target --principals=\"host/sshserver.cockpit.lan@COCKPIT.LAN\"'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add cockpit-delegation'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add-member cockpit-delegation --principals=\"cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN\"'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add-target cockpit-delegation --servicedelegationtargets=\"cockpit-target\"'")
```

Comment:
Our services image has the [latest freeipa-server docker image](https://hub.docker.com/r/freeipa/freeipa-server/tags) from 12 days ago. Does the absence of the ldap patching hack mean that this has the expected delegation fix now?

Reply:
yup